snuba/query/conditions.py

```diff
@@ -313,40 +315,58 @@ def is_condition(exp: Expression) -> bool:
 
 
 def build_match(
```

Comment:
It's not clear to me what this function does, a high level description would be useful to orient me as a reviewer. Also this seems to be a core function and its functionality has changed but I don't see any tests of the function specifically in this PR.